api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -172,6 +155,16 @@ private long getWorkspaceId(String workspaceNamespace, String workspaceId) {
     Workspace dbWorkspace = workspaceService.getRequired(workspaceNamespace, workspaceId);
     return dbWorkspace.getWorkspaceId();
   }
-}
 
+  private FileDetail convertStringToFileDetail(String str) {
+    if (str != null && str.startsWith("gs://")) {
+      int pos = str.lastIndexOf('/') + 1;
+      String fileName = str.substring(pos);
+      String filePath = str.replaceFirst(fileName + "$", "");
```

Comment:
Perhaps use `Matcher.QuoteReplacement` on fileName here. Could be problems if I decided to name my notebook `.*.ipynb`

Reply:
👍

Reply:
Done.